Add contributing file (#36)
thedataquarry/rustinpieces: CONTRIBUTING.md

> Thank you for your interest in contributing to the Rust In Pieces!

Reviewer: ```suggestion
Thank you for your interest in contributing to the *Rust In Pieces* project!
```

There are a few ways of helping: critiquing the documentation and/or code examples, fixing
incorrect information, and fixing bugs.

> It is encouraged for you to run the tests, formatters, and linters before submitting PRs. They will

Reviewer: I agree with you, adding the Rust/Python formatting guidelines are necessary because a lot of people will never have worked `pre-commit.yml` and the likes before (and they may not even have it installed).

Author: I added a section on using pre-commit. Let me know if these instructions explain it well for a new user to pre-commit.

all need to pass before the PR can be merged so you can save youself some time by running them
locally before submitting.

## Critiquing the documentation and/or code

This is the easiest way to contribute. Basically, as you read the documentation or experiment with
the code samples, if you find something confusing, incorrect, or missing, then you can file an
[issue](https://github.com/thedataquarry/rustinpieces/issues) explaining your concerns.

## Fixing incorrect information

If you find typos or incorrect links we welcome PRs to fix them. For these types of fixes there is

> no need to open an issue first. For larger changes you are encouraged to first open an issue to

Reviewer: ```suggestion
no need to open an issue first. For larger changes (or potentially adding a new piece), you are encouraged to first open an issue to
```

discuss the change before taking the time to make changes.

## Fixing bugs

If you find a bug in the code examples as you work through them, PRs for fixes are welcome. When
fixing a bug in either a Python or Rust file please check the other language's corresponding file

> to see if the same bug exists there. If you now how to fix both then you can do both in one PR. If

Reviewer: ```suggestion
to see if the same bug exists there. If you know how to fix both then you can do them both in one PR. If
```
